docs/components/plugins/api/APISectionUtils.tsx

```diff
@@ -258,6 +259,14 @@ export const resolveTypeName = (
       }
       return elementType.name + type;
     } else if (type === 'union' && types?.length) {
+      const isLargeLiteralUnion =
+        types.length > LITERAL_UNION_COLLAPSE_THRESHOLD &&
+        types.every((t: TypeDefinitionData) =>
+          ['literal', 'templateLiteral', 'intrinsic', 'reference', 'tuple'].includes(t.type)
+        );
+      if (isLargeLiteralUnion) {
+        return 'See description for available icons.';
```

Comment:
Should this message be hardcoded? Maybe it's worth adding a check wether the name includes `Icon`?

We could do something like:
```
if(isIcon) { .... }
else { console.warn('The type literal is long. Consider adding a custom message') } 
```